src/cluster_import.c

```diff
@@ -0,0 +1,1927 @@
+/*
+ * Copyright Valkey Contributors.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD 3-Clause
+ */
+
+#include "server.h"
+#include "cluster.h"
+#include "cluster_legacy.h"
+#include "bio.h"
+#include "module.h"
+#include "functions.h"
+
+int isSlotMigrationLinkInProgress(slotMigrationLink *link);
+slotMigrationLink *createSlotImportLink(clusterNode *source_node, list *slot_ranges, int one_shot);
+void connectSlotImportLink(slotMigrationLink *link, char *ip, int port);
+const char *slotMigrationLinkStateToString(slotMigrationLinkState state);
+const char *slotExportStateToString(slotMigrationLinkState state);
+void updateSlotMigrationLinkState(slotMigrationLink *link, slotMigrationLinkState state);
+void sendSyncSlotsMessageOnClient(client *c, const char *subcommand);
+void sendSyncSlotsMessageOnLink(slotMigrationLink *link, const char *subcommand);
+void proceedWithSlotMigration(slotMigrationLink *link);
+slotMigrationLink *createSlotExportLink(client *c, char *nodename, char *linkname, list *slot_ranges);
+int isSlotExportPauseTimedOut(slotMigrationLink *link);
+void resetSlotMigrationLink(slotMigrationLink *link);
+void finishSlotMigrationLink(slotMigrationLink *link, slotMigrationLinkState state, char *message);
+void updateSlotMigrationLinkStatusMessage(slotMigrationLink *link, char *message);
+
+void freeSlotRangeValue(void *o) {
+    zfree(o);
+}
+
+list *createSlotRangeList(void) {
+    list *slot_ranges = listCreate();
+    listSetFreeMethod(slot_ranges, freeSlotRangeValue);
+    return slot_ranges;
+}
+
+sds representSlotRangeList(list *slot_ranges, const char separator) {
```

Comment:
```suggestion
sds representSlotRangeList(list *slot_ranges) {
```
This function only takes in '-' as a separator.

Reply:
Hmm, guess you are right :)